internal/builders/istio.go

```diff
@@ -113,6 +113,13 @@ func (r *Rule) RuleFrom(val *RuleFrom) *Rule {
 	return r
 }
 
+func (r *Rule) RuleFromConditional(val *RuleFrom, condition bool) *Rule {
```

Comment:
I think having a conditional in the builder is not a good thing. I this case we make it generic, but that makes it actually harder to follow in what case this is relevant. I think as we have a builder we can move the conditional closer to the actual logic and only invoke the RuleFrom conditional, e.g. 

in generateAuthorizationPolicySpec
```
	ruleBuilder := builders.RuleBuilder().RuleTo(builders.RuleToBuilder().
		Operation(builders.OperationBuilder().Methods(rule.Methods).Path(rule.Path)))

	if processing.IsJwtSecured(rule) {
		ruleBuilder.RuleFrom(builders.RuleFromBuilder().Source())
	}

```

Reply:
Yes, agree. I was also unsure about having a conditional param within builders.